draft-wussler-openpgp-pqc.md

```diff
@@ -1565,84 +1489,88 @@ setting, the adversary has to distinguish a set of challenge keys
 
 they have a zero advantage over guessing.
 
 The only multi-user advantage that the adversary could gain therefore consists
 of queries to H that are meaningful for two different users u1 != u2 and their
-associated public keys.
-This is only the case if
+associated public keys. This is only the case if
 
     (c1*_u1, c2*_u1) = (c1*_u2, c2*_u2)
 
 as the ciphertext values decide for which challenge the query is meaningful.
-This means that a ciphertext collision is needed between challenges.
-Assuming that the randomness used in the generation of the two challenges is
+This means that a ciphertext collision is needed between challenges. Assuming
+that the randomness used in the generation of the two challenges is
 uncorrelated, this is negligible.
 
 In consequence, the ciphertexts already work sufficiently well as
 domain-separator.
 
-## SPHINCS+ {#sphincs-sec-cons}
+## SLH-DSA Message Randomizer {#slhdsa-sec-cons}
 
-The original specification of SPHINCS+ {{SPHINCS-Subm}} prescribes an optional
-randomized hashing. This is not used in this specification, as OpenPGP v6
-signatures already provide a salted hash of the appropriate size.
+The specification of SLH-DSA {{FIPS-205}} prescribes an optional
+non-deterministic message randomizer. This is not used in this specification,
+as OpenPGP v6 signatures already provide a salted signature data digest of the
+appropriate size.
 
 ## Binding hashes in signatures with signature algorithms
 
 In order not to extend the attack surface, we bind the hash algorithm used for
-message digestion to the hash algorithm used internally by the signature
-algorithm. Dilithium internally uses a SHAKE256 digest, therefore we require
-SHA3 in the Dilithium + ECC signature packet. In the case of SPHINCS+ the
-internal hash algorithm varies based on the algorithm and parameter ID.
+signature data digestion to the hash algorithm used internally by the signature
+algorithm.
+
+ML-DSA internally uses a SHAKE256 digest, therefore we require SHA3 in the
+ML-DSA + ECC signature packet, see {{mldsa-sig-data-digest}}.
+
+In the case of SLH-DSA the internal hash algorithm varies based on the
+algorithm and parameter ID, see {{slhdsa-sig-data-digest}}.
 
 
 # Additional considerations
 
-## Performance Considerations for SPHINCS+ {#performance-considerations}
+## Performance Considerations for SLH-DSA {#performance-considerations}
 
-This specification introduces both Dilithium + ECC as well as SPHINCS+ as
-PQ(/T) signature schemes.
+This specification introduces both ML-DSA + ECC as well as SLH-DSA as PQ(/T)
+signature schemes.
 
-Generally, it can be said that Dilithium + ECC provides a performance in terms
-of execution time and space requirements that is close to that of traditional
-ECC signature schemes. Implementers may want to offer SPHINCS+ for applications
+Generally, it can be said that ML-DSA + ECC provides a performance in terms of
+execution time and space requirements that is close to that of traditional ECC
```

Comment:
Hmm, I think regarding execution time of ML-DSA the statement could be held up, since according to our comparison it faster then conventional curves but considerably slower than EdDSA.

But with respect to space requirements, I don't think the statement can be held: For example for 128-bit security, Dilithium signatures are about 37x larger than EC signatures, while SLH-DSA small parameters only bring another factor of 3. 

If we compare ML-DSA with RSA (3072 bit), then ML-DSA signatures are 6x larger.

```suggestion
execution time requirements that is close to that of traditional ECC signature schemes.
Regarding the size of signatures and public keys, though, ML-DSA has far greater requirements than traditional schemes like EC-based or even RSA
```